okhttp/src/main/java/com/squareup/okhttp/Connection.java

```diff
@@ -166,13 +173,14 @@ void connect(int connectTimeout, int readTimeout, int writeTimeout, Request tunn
    * Connects this connection if it isn't already. This creates tunnels, shares
    * the connection with the connection pool, and configures timeouts.
    */
-  void connectAndSetOwner(OkHttpClient client, Object owner, Request request) throws IOException {
+  void connectAndSetOwner(OkHttpClient client, Object owner, Request request)
+      throws RouteException {
     setOwner(owner);
 
     if (!isConnected()) {
-      Request tunnelRequest = tunnelRequest(request);
-      connect(client.getConnectTimeout(), client.getReadTimeout(),
-          client.getWriteTimeout(), tunnelRequest);
+      List<ConnectionSpec> connectionSpecs = client.getConnectionSpecs();
```

Comment:
I will return it to the Address, which means the "connection specs that could be used" becomes part of the connection pooling key and anything else that keys off of Address or things that hold it. e.g. Route, Connection.

I'm unsure whether that's actually a good thing or not for all cases. Route seems like it should be independent of the ConnectionSpecs to me with this change, but I can see why Address might want to have it which means something about the design is incorrect.